### PR TITLE
Save and continue

### PR DESCRIPTION
**Pull request by bbridge0200**
Working on 
1. writing to csv on exit will null values for unannotated images (done)
2. a popup create annotation GUI
3. rendering json data from popup and from an imported csv -> annotations gui

### CODE REVIEW

File: napari_allencell_annotator/view/annotator_view.py [hunk 1]
```diff
@@ -271,10 +279,11 @@ def render_annotations(self, data: Dict[str, Dict[str, str]]):
             The dictionary of annotation types.
         """
         self.annotation_item_widgets = []
-        if len(data) < 9:
-            self.annot_list.setMaximumHeight(45.5 * len(data))
+
         for name in data.keys():
             self._create_annot(name, data[name])
+        if len(data) < 9:
+            self.annot_list.setMaximumHeight(self.annot_list.item(0).sizeHint().height() * len(data))
```
Comment (aditnath) [inline napari_allencell_annotator/view/annotator_view.py]: Might want to make this a bit more dynamic like brian was saying?

File: napari_allencell_annotator/controller/annotator_controller.py [hunk 2]
```diff
@@ -86,7 +90,8 @@ def start_annotating(self, num_images: int):
         num_images : int
             The total number of images to be annotated.
         """
-
+        for path, lst in dct.items():
```
Comment (yrkim98) [inline napari_allencell_annotator/controller/annotator_controller.py]: You're unpacking `dct` to create the same dictionary `self.annotation_dict`. You can just do `self.annotation_dct = dct`

File: napari_allencell_annotator/controller/annotator_controller.py [hunk 3]
```diff
@@ -105,17 +110,16 @@ def set_curr_img(self, curr_img: Optional[Dict[str, str]] = None):
         if curr_img is not None:
             self.curr_img = curr_img
             path: str = curr_img["File Path"]
-            if path not in self.annotation_dict.keys():
-                self.annotation_dict.update(
-                    {path: [curr_img["File Name"], curr_img["FMS"]]}
-                )
+            if len(self.annotation_dict[path]) < 3:
```
Comment (yrkim98) [inline napari_allencell_annotator/controller/annotator_controller.py:111]: Would be good to comment this. Its kind of confusing so can you explain the `if else`? Don't know what's going on when `len(self.annotation_dict[path]) < 3`.

**Review by aditnath — APPROVED** [review]
Looks really great, I see no issues with the annotation block addition. Making it more dynamic and possibly making it collapsible/a popup will be important later, but we can deal with that later.

File: napari_allencell_annotator/controller/annotator_controller.py [hunk 4]
```diff
@@ -105,17 +110,16 @@ def set_curr_img(self, curr_img: Optional[Dict[str, str]] = None):
...
+
                 self.view.render_default_values()
             else:
                 self.view.render_values(self.annotation_dict[path][2::])
-            self.view.set_curr_index(curr_img["Row"])
+            self.view.set_curr_index(int(curr_img["Row"]))
```
Comment (yrkim98) [inline napari_allencell_annotator/controller/annotator_controller.py:120]: what does curr_img store?

File: napari_allencell_annotator/widgets/list_widget.py [hunk 5]
```diff
@@ -44,11 +42,10 @@ def __init__(self):
         QListWidget.__init__(self)
         self.checked: Set[ListItem] = set()
         self.setSelectionMode(QAbstractItemView.SingleSelection)
-        self.files: Set[str] = set()
-        self.file_order: List[str] = []
+        self.file_dict: Dict[str, List[str]] = {}
         self.setCurrentItem(None)
         self._shuffled: bool = False
-        self.shuffle_order: List[str] = []
+        self.shuffle_order: Dict[str, Dict[str, str]] = {}
```
Comment (yrkim98) [inline napari_allencell_annotator/widgets/list_widget.py:54]: What do these dicts store?

**Review by yrkim98 — CHANGES_REQUESTED** [review]
Nice work Beatrice. Other than the one performance fix, rest of the issues are just related to code readability (don't be afraid to use more comments!)

File: napari_allencell_annotator/controller/annotator_controller.py (hunk 4 above)
Comment (bbridge0200) [reply napari_allencell_annotator/controller/annotator_controller.py:120]: Putting a bunch of comments on this method . curr_img = {'File Path' : 'path', 'Row' : str(rownum)}

File: napari_allencell_annotator/widgets/list_widget.py (hunk 5 above)
Comment (bbridge0200) [reply napari_allencell_annotator/widgets/list_widget.py:54]: # file_dict holds all image info file path -> [file name, FMS] 
# also holds the original insertion order in .keys()
self.file_dict: Dict[str, List[str]] = {}

# shuffle order holds the same file path -> [file name, FMS] as file_dict
# only filled when the images have been shuffled
# holds a new shuffled order in .keys()
# when annotation starts if images are shuffled this order is given to annotation view
self.shuffle_order: Dict[str, Dict[str, str]] = {}

Let me know if you think that this system makes sense! I need both orders for the un-shuffle feature and shuffle order is cleared out when the data is unshuffled

Comment (bbridge0200) [reply napari_allencell_annotator/widgets/list_widget.py:54]: sorry for the weird formatting, didn't like hashtags apparently

**Pull request merged**